### PR TITLE
Use badger brain as data source

### PR DESCRIPTION
**Pull request by lpil**
head-bop

- [x] Content in prismic on staging
- [ ] Content in prismic on prod
- [ ] Badger Brain deployed to prod
- [x] Badger brain data source
- [x] Dummy data source for tests
- [x] Use BB data source in prod
- [x] Community ID and BB endpoint are configurable
- [x] Featured event details pulling through
- [x] Speaker images
- [x] Router tests for happy path
- [x] Tested on staging

### CODE REVIEW

File: server/data/badger-brain-test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+} from './badger-brain';
+
+describe('data/badger-brain backend getSiteState', () => {
+  it('uses the http client to query badger brain', done => {
+    const postFn = (url, query) => {
+      expect(url).to.equal(badgerBrainURL);
+      expect(query).to.equal(siteStateQuery);
+      done();
+    };
+    getSiteState(postFn);
```
Comment (bkspace) [inline server/data/badger-brain-test.js]: You can omit the done callback and just return 'getSiteState(postFn)' - preference?

Comment (lpil) [inline server/data/badger-brain-test.js]: You can? Cool.

Comment (lpil) [inline server/data/badger-brain-test.js]: Hey, this is much nicer. Thanks :)

File: shared/components/NextCommunityEvent/index.js [hunk 2]
```diff
@@ -12,9 +13,19 @@ const calendarURL = 'https://calendar.google.com/calendar/event?action=TEMPLATE'
   '&tmeid=NWY0cDE3Y3N0MzZhbWp2amxmdjhkdHBqbGsgbG5kaDVzdXRrbmtyZjZpbjEzYWgzYmUwbW9AZw' +
   '&tmsrc=lndh5sutknkrf6in13ah3be0mo%40group.calendar.google.com';
 
-const locationURL = 'https://goo.gl/maps/Z8SU87i4Fy42';
+function googleMapsUrl(location) {
+  const { latitude, longitude } = pathOr({}, ['coordinates'], location);
+  if (!latitude && !longitude) { return null; }
```
Comment (AndrewBestbier) [inline shared/components/NextCommunityEvent/index.js]: Or instead of and? If either are undefined we don't want to generate a gmap link.

Comment (lpil) [inline shared/components/NextCommunityEvent/index.js]: Sure. The styling still makes it look like a link though.

File: server/router/test.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { useDummyData } from '../../test/mock-data-source';
 
 describe('GET /__health__', () => {
   it('returns 200, to be checked by load balancer', () => {
-    request(app)
+    return request(app)
```
Comment (lpil) [inline server/router/test.js]: Probably less error prone to remove the `{}`s as it's a single expression.

Handling async tests like this seems *really* fragile, easy to accidentally not return anything. Is there a better way to do this?

**Pull request merged**